### DISCUSSION TITLE
BitOps Community Meeting (26 Oct 2022): Community Metrics, Pylint, General Roadmap

### DISCUSSION THREAD

**Discussion by arm4b** [bitovi/bitops#331] [Announcements]
We invite everyone (developers, adopters, or just curious people) to join the bi-weekly BitOps Community `1 hour` Meeting on Wednesday, `26 Oct 2022, 12:00 PM US East` to discuss the ongoing project developments, updates, and plans, share use cases and stories or ask/answer questions about BitOps tool.

See https://github.com/bitovi/bitops/discussions/288 for more info on how to join.

### Follow-ups from the previous #319 Meeting (10min)
- BitOps landing page (@mickmcgrath13 @armab)
  - BitOps 1-3min intro Video (@PhillypHenning)
- Verify & Update [BitOps HeyEmoji blog](https://www.bitovi.com/blog/heyemoji-slack-app-aws-terraform-ansible) to v2 (@armab)
- Verify & Update [Combine Terraform and Ansible to Provision and Configure a Web Server](https://www.bitovi.com/blog/bitops-terraform-ansible) to v2 (@LeoDiazL)
- Trying Twitch: BitOps HeyEmoji demo (@dylan-bitovi, @ccapell, @PhillypHenning)

### Discussion: How do we know how many users use BitOps? (15min)
In one of the previous meetings, @ccapell asked a question how do we know how many users rely on BitOps?
Discuss the tools and metrics that would help us to answer this question.

### Black & Pylint Showcase (15min)
* In https://github.com/bitovi/bitops/pull/297 we've added a tool called `black` (https://black.readthedocs.io/en/stable/) that allows reformatting the code to a common python style (like `go fmt`).
* In https://github.com/bitovi/bitops/pull/298 we're adding a python static code analyzer called `pylint` (https://pylint.pycqa.org/en/latest/) that encourages some good practices.
Both become part of the checks enforced by CI in the Bitops project.

Highlight the changes in more detail, how to use these tools and how they help.

### General Roadmap Overview (15min)
Highlight the conclusions and goals based on a non-technical Roadmap planning for the next 3mo and 6mo.

### Development: Random Issue
If time allows, pick up one of the features from the (Development Roadmap)[https://github.com/bitovi/bitops/discussions/245] and discuss it: design, implementation, UX, ideas, technical details.

**Comment by mickmcgrath13** [comment]
Call Notes

# Last time: 
- roadmap

## Follow ups
- BitOps landing page
- quick video (@PhillypHenning)
- 2 blog posts to update for v2 (heyemoji: @armab, aws terraform helm, @LeoDiazL)
- @LeoDiazL 
  - starting to take shape. some little issues/discrepancies
  - there is a learning curve. there are a lot of tools that require some expertise (aws cli, tf, helm, ansible). need a little knowledge of each to start. hardest part for anyone to get started.  It's good and promising, but has a lot of moving parts.
  - Should get more examples for anyone to come in to grab and get started with
  - https://github.com/bitovi/bitops/issues/321
- Twitch
  - scheduling heyemoji at some point soon

# This time:
- rename to BitOps Office Hours
- Videos are generally ranked higher. should have a video for blog posts
- community metrics
- new tools (black, pylint)
- general roadmap

## How do we know how many users use BitOps?
- can answer indirectly by measuring community metrics/analytics
  - slack, github, social, visits to the blogs, interactions with users, dockerhub downloads, etc
  - https://github.com/bitovi/bitops/graphs/traffic
  - https://chaoss.community/
  - Maybe a github action for bitops
  - Tracking plugin?
    - a different plugin that we use vs external users

## New tools

- [Black](https://github.com/bitovi/bitops/pull/297)
  - it's pretty simple. 
  - enforces common code formatting
  - "black" reference to one of Henry Ford's early cars. "can have a car painted any color as long as it's black"
  - enforced by the codebase. when you add python, it's checked by this tool
  - if error, you have to run it locally to reformat ([example](https://github.com/bitovi/bitops/actions/runs/3323834010/jobs/5494675369))
  - or you can use the suggested change (it shows a diff for a suggested change)
- [Pylint](https://github.com/bitovi/bitops/pull/298)
 
## Roadmap
- [Technical roadmap](https://github.com/bitovi/bitops/projects/6)
- Non-technical roadmap (bitovi internal)
  - 3 phases
    - 1 (3mo - done) - Research
    - 2 (3mo - current) - Improvements (docs, quality, intuitiveness, guides, etc)
      - blocker for community to contribute back. if they look deeper in the project, and we don't have good quality, they won't extend it or contribute
    - 3 (3mo - coming up) - Marketing
  - There will be some overlap between 2 and 3
